src/shared/components/navbar/SearchBox.js

```diff
@@ -9,30 +9,41 @@ import styles from "~/shared/styles/navbar";
 @Radium
 export default class SearchBox extends React.Component {
 
-    constructor(props) {
-        super(props);
-
-        this.state = {
-            query: "",
-            userIds: []
-        }
+    state = {
+        query: "",
+        users: [],
+        team: {},
     }
 
     sendQuery = async(query) => {
         if (query == "") {
             this.setState({
-                userIds: []
+                users: [],
+                team: {},
             });
         } else {
             try {
                 let { data } = await ajax.request("get", "/users/search?search=" + query);
                 this.setState({
-                    userIds: data,
+                    users: data,
                 });
             } catch (err) {
                 console.log(err);
             }
-            //TODO: make this faster
+            if ((/^\d+$/).test(query)) {
+                try {
+                    let teamInfo = await ajax.request("get", "/teams/number/" + query + "/info");
```

Comment:
If you type two digits into the search bar in quick succession, sometimes the second request will finish before the first request and the wrong team number will be shown. This is a perfect use case for cancellable requests, where were added and then removed. The history can still be seen with commit d9f5d94.